{"name": "잘못된 ID,PW 입력 시 로그인이 실패하는지 확인", "status": "failed", "statusDetails": {"message": "Failed: ❌ 로그인 실패 :Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"pw\"]\"}\n  (Session info: chrome=134.0.6998.165); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x5568defeaffa <unknown>\n#1 0x5568deaa9970 <unknown>\n#2 0x5568deafb385 <unknown>\n#3 0x5568deafb5b1 <unknown>\n#4 0x5568deb4a3c4 <unknown>\n#5 0x5568deb212bd <unknown>\n#6 0x5568deb4770c <unknown>\n#7 0x5568deb21063 <unknown>\n#8 0x5568deaed328 <unknown>\n#9 0x5568deaee491 <unknown>\n#10 0x5568defb242b <unknown>\n#11 0x5568defb62ec <unknown>\n#12 0x5568def99a22 <unknown>\n#13 0x5568defb6e64 <unknown>\n#14 0x5568def7dbef <unknown>\n#15 0x5568defd9558 <unknown>\n#16 0x5568defd9736 <unknown>\n#17 0x5568defe9e76 <unknown>\n#18 0x7f1e1489caa4 <unknown>\n#19 0x7f1e14929c3c <unknown>", "trace": "driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"6c5893d915605f6647d4f5f48dd72d9b\")>\n\n    @allure.feature(\"예외 케이스\")\n    @allure.story(\"잘못된 ID,PW 입력\")\n    @allure.title(\"잘못된 ID,PW 입력 시 로그인이 실패하는지 확인\")\n    def test_login(driver):\n        driver.get(\"https://www.kurly.com/main\")\n        driver.maximize_window()\n    \n        try:\n            driver.get(\"https://www.kurly.com/main\")\n            time.sleep(2)\n            login_button = driver.find_element(By.XPATH, \"//a[contains(text(),'로그인')]\")\n            login_button.click()\n            time.sleep(2)\n    \n            #아이디 패스워드 입력\n            login_username_input = driver.find_element(By.NAME, \"id\")\n>           login_password_input = driver.find_element(By.ID, \"pw\")\n\nsrc/tests/ui_tests/test_ui_login.py:23: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:770: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:384: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7fb9fffb3c90>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#17 0x5568defe9e76 \\\\u003Cunknown>\\\\n#18 0x7f1e1489caa4 \\\\u003Cunknown>\\\\n#19 0x7f1e14929c3c \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"pw\"]\"}\nE         (Session info: chrome=134.0.6998.165); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x5568defeaffa <unknown>\nE       #1 0x5568deaa9970 <unknown>\nE       #2 0x5568deafb385 <unknown>\nE       #3 0x5568deafb5b1 <unknown>\nE       #4 0x5568deb4a3c4 <unknown>\nE       #5 0x5568deb212bd <unknown>\nE       #6 0x5568deb4770c <unknown>\nE       #7 0x5568deb21063 <unknown>\nE       #8 0x5568deaed328 <unknown>\nE       #9 0x5568deaee491 <unknown>\nE       #10 0x5568defb242b <unknown>\nE       #11 0x5568defb62ec <unknown>\nE       #12 0x5568def99a22 <unknown>\nE       #13 0x5568defb6e64 <unknown>\nE       #14 0x5568def7dbef <unknown>\nE       #15 0x5568defd9558 <unknown>\nE       #16 0x5568defd9736 <unknown>\nE       #17 0x5568defe9e76 <unknown>\nE       #18 0x7f1e1489caa4 <unknown>\nE       #19 0x7f1e14929c3c <unknown>\n\nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:232: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"6c5893d915605f6647d4f5f48dd72d9b\")>\n\n    @allure.feature(\"예외 케이스\")\n    @allure.story(\"잘못된 ID,PW 입력\")\n    @allure.title(\"잘못된 ID,PW 입력 시 로그인이 실패하는지 확인\")\n    def test_login(driver):\n        driver.get(\"https://www.kurly.com/main\")\n        driver.maximize_window()\n    \n        try:\n            driver.get(\"https://www.kurly.com/main\")\n            time.sleep(2)\n            login_button = driver.find_element(By.XPATH, \"//a[contains(text(),'로그인')]\")\n            login_button.click()\n            time.sleep(2)\n    \n            #아이디 패스워드 입력\n            login_username_input = driver.find_element(By.NAME, \"id\")\n            login_password_input = driver.find_element(By.ID, \"pw\")\n            login_submit_button = driver.find_element(By.ID,\"log.login\")\n            time.sleep(2)\n    \n            login_username_input.clear()\n            login_password_input.clear()\n    \n            login_username_input.send_keys('boksoon8')\n            login_password_input.send_keys('1232133123')\n            login_submit_button.click()\n            time.sleep(3)\n    \n        except Exception as e:\n            capture_screenshot(driver, \"로그인\", \"screenshots_login\")\n>           pytest.fail(f\"❌ 로그인 실패 :{str(e)}\")\nE           Failed: ❌ 로그인 실패 :Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"[id=\"pw\"]\"}\nE             (Session info: chrome=134.0.6998.165); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE           Stacktrace:\nE           #0 0x5568defeaffa <unknown>\nE           #1 0x5568deaa9970 <unknown>\nE           #2 0x5568deafb385 <unknown>\nE           #3 0x5568deafb5b1 <unknown>\nE           #4 0x5568deb4a3c4 <unknown>\nE           #5 0x5568deb212bd <unknown>\nE           #6 0x5568deb4770c <unknown>\nE           #7 0x5568deb21063 <unknown>\nE           #8 0x5568deaed328 <unknown>\nE           #9 0x5568deaee491 <unknown>\nE           #10 0x5568defb242b <unknown>\nE           #11 0x5568defb62ec <unknown>\nE           #12 0x5568def99a22 <unknown>\nE           #13 0x5568defb6e64 <unknown>\nE           #14 0x5568def7dbef <unknown>\nE           #15 0x5568defd9558 <unknown>\nE           #16 0x5568defd9736 <unknown>\nE           #17 0x5568defe9e76 <unknown>\nE           #18 0x7f1e1489caa4 <unknown>\nE           #19 0x7f1e14929c3c <unknown>\n\nsrc/tests/ui_tests/test_ui_login.py:37: Failed"}, "attachments": [{"name": "Failure Screenshot", "source": "efe0ffe4-5be9-4bdf-8cdd-b2f059d6774f-attachment.png", "type": "image/png"}, {"name": "Failure Screenshot", "source": "c969fb99-a3fb-4462-a5a2-67cc689fa2df-attachment.png", "type": "image/png"}, {"name": "stdout", "source": "86053589-8b45-4c38-8133-1bac8fa73219-attachment.txt", "type": "text/plain"}], "start": 1744130067039, "stop": 1744130074178, "uuid": "6f9481e8-3c9b-4728-bb09-6dced840a892", "historyId": "98e8f149c7227721782a965f730ba74a", "testCaseId": "98e8f149c7227721782a965f730ba74a", "fullName": "src.tests.ui_tests.test_ui_login#test_login", "labels": [{"name": "story", "value": "잘못된 ID,PW 입력"}, {"name": "feature", "value": "예외 케이스"}, {"name": "parentSuite", "value": "src.tests.ui_tests"}, {"name": "suite", "value": "test_ui_login"}, {"name": "host", "value": "fv-az1674-192"}, {"name": "thread", "value": "2603-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "src.tests.ui_tests.test_ui_login"}]}